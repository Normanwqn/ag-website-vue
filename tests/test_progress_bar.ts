import { mount } from '@vue/test-utils';

import ProgressBar from '@/components/progress_bar.vue';

describe('Progress bar tests', () => {
    test('Progress 0', () => {
        const wrapper = mount(ProgressBar, {
            propsData: {
                progress: 0
            }
        });

        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    test('Partial progress', () => {
        const wrapper = mount(ProgressBar, {
            propsData: {
                progress: 51
            }
        });

        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    test('Progress 100% default slot', () => {
        const wrapper = mount(ProgressBar, {
            propsData: {
                progress: 100
            }
        });

        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    test('Progress 100% custom slot', () => {
        const wrapper = mount(ProgressBar, {
            propsData: {
                progress: 100
            },
            slots: {
                default: 'Finishing request'
            }
        });

        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});
