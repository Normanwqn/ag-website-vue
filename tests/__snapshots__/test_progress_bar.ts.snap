// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress bar tests Partial progress 1`] = `
<div
  id="total-progress"
>
  <div
    id="completed-progress"
    style="width: 51%;"
  >
    <div
      id="progress-text"
    >
      51%
    </div>
  </div>
</div>
`;

exports[`Progress bar tests Progress 0 1`] = `
<div
  id="total-progress"
>
  <div
    id="completed-progress"
    style="width: 0%;"
  >
    <div
      id="progress-text"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress bar tests Progress 100% custom slot 1`] = `
<div
  id="total-progress"
>
  <div
    id="completed-progress"
    style="width: 100%;"
  >
    <div
      id="progress-text"
    >
      Finishing request
    </div>
  </div>
</div>
`;

exports[`Progress bar tests Progress 100% default slot 1`] = `
<div
  id="total-progress"
>
  <div
    id="completed-progress"
    style="width: 100%;"
  >
    <div
      id="progress-text"
    >
      
          Please wait...
        
    </div>
  </div>
</div>
`;
